cloud-openstack-server/src/main/java/jetbrains/buildServer/clouds/openstack/CreateImageOptions.java

```diff
@@ -0,0 +1,157 @@
+
+package jetbrains.buildServer.clouds.openstack;
+
+import java.util.concurrent.ScheduledExecutorService;
+
+import org.jclouds.openstack.nova.v2_0.options.CreateServerOptions;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import jetbrains.buildServer.serverSide.ServerPaths;
+import jetbrains.buildServer.util.StringUtil;
+
+public class CreateImageOptions {
+
+    @NotNull
+    private OpenstackApi openstackApi;
+    @NotNull
+    private String imageId;
+    @NotNull
+    private String imageName;
+    @NotNull
+    private String openstackImageName;
+    @NotNull
+    private String flavorName;
+    @Nullable
+    private String volumeName;
+    @Nullable
+    private String volumeDevice;
+    @NotNull
+    private boolean autoFloatingIp;
+    @NotNull
+    private CreateServerOptions createServerOptions;
+    @Nullable
+    private String userScriptPath;
+    @NotNull
+    private ServerPaths serverPaths;
+    @NotNull
+    private ScheduledExecutorService scheduledExecutorService;
+
+    protected CreateImageOptions openstackApi(@NotNull final OpenstackApi openstackApi) {
+        this.openstackApi = openstackApi;
+        return this;
+    }
+
+    protected CreateImageOptions imageId(@NotNull final String imageId) {
+        this.imageId = imageId;
+        return this;
+    }
+
+    protected CreateImageOptions imageName(@NotNull final String imageName) {
+        this.imageName = imageName;
+        return this;
+    }
+
+    protected CreateImageOptions openstackImageName(@NotNull final String openstackImageName) {
+        this.openstackImageName = openstackImageName;
+        return this;
+    }
+
+    protected CreateImageOptions flavorName(@NotNull final String flavorName) {
+        this.flavorName = flavorName;
+        return this;
+    }
+
+    /**
+     * Volume should be "volumeName,volumeDevice"
+     * 
+     * @param volume volume name and volume device
+     * @return CreateImageOptions
+     */
+    protected CreateImageOptions volume(@Nullable final String volume) {
+        if (StringUtil.isNotEmpty(volume)) {
+            String[] volumeNameDevice = volume.split(",");
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: A "NullPointerException" could be thrown; "volume" is nullable here. ([squid:S2259](https://sonarcloud.io/organizations/axel3rd-github/rules#rule_key=squid%3AS2259))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=jetbrains.buildServer.clouds%3Acloud-openstack&pullRequest=32&issues=AWT_zzZ9NS0RWzbYcPPe&open=AWT_zzZ9NS0RWzbYcPPe)

[ISSUE_KEY]: AWT_zzZ9NS0RWzbYcPPe